src/lib/work_partitioner/job_pool.ml

```diff
@@ -0,0 +1,125 @@
+open Core_kernel
+
+type ('accum, 'final) fold_action =
+  { action : [ `Continue of 'accum | `Stop of 'final ]; slashed : bool }
```

Comment:
What is `slashed`? Is it like `remove_item`?

If yes, convention used in other parts of code (and libraries) is to have four constructors:

* `Continue`
* `Continue_remove`
* `Stop`
* `Stop_remove`

and then there is no need for a new structure type, also semantics become slightly more obvious (though this one is up to one's preferences)